src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -0,0 +1,83 @@
+package cromwell.engine
+
+import java.util.UUID
+
+import akka.actor.{Actor, ActorSystem, ActorRef}
+import akka.pattern.pipe
+import scala.concurrent.ExecutionContext.Implicits.global
+import cromwell.binding.{WdlBinding, WdlInputs, Wdl}
+import cromwell.engine.WorkflowActor.Start
+import cromwell.engine.WorkflowManagerActor.{SubmitWorkflow, WorkflowStatus}
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.local.LocalBackend
+
+import scala.concurrent.Future
+
+/**
+ * Abstract notion of a workflow manager. Mainly exists to allow for actor-system-free unit testing of basic concepts
+ */
+trait WorkflowManager {
+  type Workflow
+
+  case class ManagedWorkflow(id: WorkflowId, workflow: Workflow)
+
+  def backend: Backend
+
+  def generateWorkflow(wdl: Wdl, inputs: WdlInputs): Workflow
+  def workflowStatus(id: WorkflowId): Option[WorkflowState]
+
+  private var managedWorkflows = Map.empty[WorkflowId, Workflow]
+
+  def submitWorkflow(wdl: Wdl, inputs: WdlInputs): Option[ManagedWorkflow] = {
+    val managedWorkflow = ManagedWorkflow(UUID.randomUUID(), generateWorkflow(wdl, inputs))
+    if (managedWorkflows contains managedWorkflow.id) None
```

Comment:
Perhaps this should use `Try` rather than `Option`?  `Option` doesn't really capture the badness of already finding a submitted workflow with this id.